lib/api/auth/authentication.dart

```diff
@@ -60,104 +56,44 @@ class Authentication {
   }
 
   // A function to login the user with email and password
-  Future<void> login(
-      String email, String password, BuildContext context) async {
-    try {
-      ///  here account is the object of Account class and create session
-      ///  is a method of Account class which signs in the current user.
-      ///  We are using try catch block so that if there is any error we can
-      ///  show the user a proper message
-      ///  if the try is successful we would be actually checking which type of
-      ///  data we are receving from the server
-      ///  if you don't want to see you can comment it out.
-      ///  nevermind I did that for you😉
-      /// var data = await account.createSession(email: email, password: password);
-      await account.createSession(email: email, password: password);
-
-      await Navigator.pushReplacementNamed(context, HomePage.routename);
-    } catch (e) {
-      // print(e);
-      await showDialog(
-          context: context,
-          builder: (BuildContext context) => AlertDialog(
-                title: const Text('Error Occured'),
-                content: Text(e.toString()),
-                actions: [
-                  TextButton(
-                      onPressed: () {
-                        Navigator.of(context).pop();
-                      },
-                      child: const Text("Ok"))
-                ],
-              ));
-    }
+  Future<void> login(String email, String password) async {
+    ///  here account is the object of Account class and create session
+    ///  is a method of Account class which signs in the current user.
+    ///  We are using try catch block so that if there is any error we can
+    ///  show the user a proper message
+    ///  if the try is successful we would be actually checking which type of
+    ///  data we are receving from the server
+    ///  if you don't want to see you can comment it out.
+    ///  nevermind I did that for you😉
+    /// var data = await account.createSession(email: email, password: password);
+    await account.createEmailSession(email: email, password: password);
```

Comment:
Wrap these functions in the `try` `catch` block please. Just be double safe with everything.
In the catch just `debuPrint` or use `log` to get the log of these blogs. No need to rethrow to block the UI

Reply:
I wrapped the function calling this `auth.signUp` &&  `auth.signIn`  with a `try` `catch` if you look. So I catch the error where the UI is actually loading and show the error there. Instead of `catch`-ing the error twice.

Reply:
Oh I get. I will change this.